numpy/tests/typing/pass/scalars.py

```diff
@@ -96,3 +96,41 @@ def __float__(self):
 np.void(np.bool_(True))
 np.void(b"test")
 np.void(np.bytes_("test"))
+
+# Test that numpy and builtin scalars can be used interchangeably
+
+def func_float(a: float) -> None:
+    pass
+
+def func_int(a: int) -> None:
+    pass
```

Comment:
As an example of why I think this is a mistake, mypy will allow this code even though it raises AttributeError:
```python
def func_int(a: int) -> None:
    return a.to_bytes(1, byteorder="little")
func_int(np.int8(1))
```

Reply:
Note that there are already a few of such cases in mypy, as `int` does not implement all methods which exists in `float`.

``` python
>>> def func(a: float) -> bool:
...     return a.is_integer()

# NOTE: mypy considers `int` a subclass of `float`
>>> func(1)
Traceback (most recent call last):
  ...
AttributeError: 'int' object has no attribute 'as_integer'
```

The number cases where the `int` / `np.integer` does work is substantially larger then those were it doesn't (just as with `int` & `float`), to the point where I feel a more pragmatic approach is justified.

Reply:
Note that the problems with `int` vs `float` are slowly being resolved by adding the missing methods (https://bugs.python.org/issue26680)